app/(auth)/signup/_components/signupForm.tsx

```diff
@@ -0,0 +1,109 @@
+'use client';
+
+import { Controller, FieldValues, useForm } from 'react-hook-form';
+
+import { AxiosError } from 'axios';
+import { useRouter } from 'next/navigation';
+
+import Button from '@/components/button/button';
+import Input from '@/components/input/input';
+import { postSignUpInfo } from '@/services/api';
+
+import MemberType from './memberType';
+
+const EMAIL_REGEX = /^[A-Za-z0-9]([-_.]?[A-Za-z0-9])*@[a-z0-9]([-_.]?[A-Za-z0-9])*\.[A-Za-z]{2,3}$/;
+const PASSWORD_REGEX = /^.{8,}$/;
+
+export default function SignUpForm() {
+  const router = useRouter();
+
+  const {
+    register,
+    control,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm<FieldValues>({ mode: 'all', defaultValues: { type: 'employee' } });
+
+  const FORM_FIELDS = [
+    {
+      name: 'email',
+      label: '이메일',
+      type: 'text',
+      placeholder: '이메일을 입력해주세요.',
+      validation: {
+        required: '이메일을 입력해주세요.',
+        pattern: {
+          value: EMAIL_REGEX,
+          message: '올바른 이메일 주소가 아닙니다.',
+        },
+      },
+    },
+    {
+      name: 'password',
+      label: '비밀번호',
+      type: 'password',
+      placeholder: '비밀번호를 입력해주세요.',
+      validation: {
+        required: '비밀번호를 입력해주세요.',
+        pattern: {
+          value: PASSWORD_REGEX,
+          message: '비밀번호를 8자 이상 입력해주세요.',
+        },
+      },
+    },
+    {
+      name: 'password_repeat',
+      label: '비밀번호 확인',
+      type: 'password',
+      placeholder: '비밀번호를 한 번 더 입력해주세요.',
+      validation: {
+        validate: (value: string) => value === watch('password') || '비밀번호가 일치하지 않습니다.',
+      },
+    },
+  ];
+
+  const onSubmit = async (data: FieldValues) => {
+    try {
+      await postSignUpInfo(data);
+      router.push('/signin');
+      alert('회원가입이 완료되었습니다.');
```

Comment:
폼을 제출하고 나서의 실행 흐름을 생각해보면 아래와 같을 것 같은데요!
1. 회원가입 api를 호출한다.
2. 회원가입 완료 메시지를 띄워준다.
3. 로그인 페이지로 이동한다.

각각은 독립된 행동이라고 저는 생각을 해서 각각 개행으로 구분지어주면 코드를 이해하기 더 좋을 것 같아요!
그리고 순서는 페이지 이동보다 alert가 먼저 나오면 좋을 것 같아요!
```suggestion
      await postSignUpInfo(data);
      
      alert('회원가입이 완료되었습니다.');
      
      router.push('/signin');
```